openhands_cli/acp_impl/agent.py

```diff
@@ -0,0 +1,600 @@
+        return AuthenticateResponse()
+
+    async def newSession(self, params: NewSessionRequest) -> NewSessionResponse:
+        """Create a new conversation session."""
+        session_id = str(uuid.uuid4())
+
+        try:
+            # Transform ACP MCP servers to Agent format
+            mcp_servers_dict = None
+            if params.mcpServers:
+                mcp_servers_dict = transform_acp_mcp_servers_to_agent_format(
+                    params.mcpServers
+                )
+
+            # Validate working directory
+            working_dir = params.cwd or str(Path.cwd())
+            logger.info(f"Using working directory: {working_dir}")
+
+            # Create conversation and cache it for future operations
+            # This reuses the same pattern as openhands --resume
+            conversation = self._get_or_create_conversation(
+                session_id=session_id,
+                working_dir=working_dir,
+                mcp_servers=mcp_servers_dict,
+            )
+
+            logger.info(
+                f"Created new session {session_id} with model: "
+                f"{conversation.agent.llm.model}"  # type: ignore[attr-defined]
+            )
+
+            return NewSessionResponse(sessionId=session_id)
+
+        except MissingAgentSpec as e:
+            logger.error(f"Agent not configured: {e}")
+            raise ValueError(
+                "Agent not configured. Please run 'openhands' to configure "
+                "the agent first."
+            )
+        except Exception as e:
+            logger.error(f"Failed to create new session: {e}", exc_info=True)
+            raise
+
+    async def prompt(self, params: PromptRequest) -> PromptResponse:
+        """Handle a prompt request.
+
+        Per ACP spec: Runs the agent until completion or a stop condition is met.
+        Returns appropriate stopReason based on why the turn ended.
+
+        This method handles the confirmation loop if confirmation policy is active.
+        """
+        session_id = params.sessionId
+
+        try:
+            # Validate session ID
+            if session_id not in self._active_sessions:
+                raise ValueError(f"Unknown session: {session_id}")
+
+            # Clear cancellation state for this session (new prompt turn)
+            self._cancelled_sessions.discard(session_id)
+
+            # Get conversation from active sessions
+            conversation = self._active_sessions[session_id]
+
+            # Convert ACP prompt format to OpenHands message content
+            message_content = convert_acp_prompt_to_message_content(params.prompt)
+
+            if not message_content:
+                return PromptResponse(stopReason="end_turn")
+
+            # Send the message with potentially multiple content types
+            # (text + images)
+            message = Message(role="user", content=message_content)
+            conversation.send_message(message)
```

Comment:
Do we want to run this in a thread? Does ACP support multiple concurrent agent chats?

Reply:
Noting that we should switch to textual library for CLI UI, which also uses asyncio - so it seems we'll use asyncio exclusively in the future and potentially can share some core logic for running the agent between the regular CLI and ACP variant